From: Gal Pressman <gal@nvidia.com>
Date: Sun, 17 Jul 2022 14:04:29 +0300
Subject: net/mlx5e: Fix wrong use of skb_tcp_all_headers() with encapsulation
Patch-mainline: v6.0-rc1
Git-commit: ec082d31c161976a0e81c5d2e3ac317237ac5a0d
References: jsc#PED-1549

Use skb_inner_tcp_all_headers() instead of skb_tcp_all_headers() when
transmitting an encapsulated packet in mlx5e_tx_get_gso_ihs().

Fixes: 504148fedb85 ("net: add skb_[inner_]tcp_all_headers helpers")
Cc: Eric Dumazet <edumazet@google.com>
Signed-off-by: Gal Pressman <gal@nvidia.com>
Reviewed-by: Tariq Toukan <tariqt@nvidia.com>
Signed-off-by: Saeed Mahameed <saeedm@nvidia.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/mellanox/mlx5/core/en_tx.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/mellanox/mlx5/core/en_tx.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/en_tx.c
@@ -137,7 +137,7 @@ mlx5e_tx_get_gso_ihs(struct mlx5e_txqsq
 	u16 ihs;
 
 	if (skb->encapsulation) {
-		ihs = skb_tcp_all_headers(skb);
+		ihs = skb_inner_tcp_all_headers(skb);
 		stats->tso_inner_packets++;
 		stats->tso_inner_bytes += skb->len - ihs;
 	} else {
